lighthouse-core/lib/url-shim.js

```diff
@@ -25,4 +25,26 @@
 
 // TODO: Add back node require('url').URL parsing when bug is resolved:
 // https://github.com/GoogleChrome/lighthouse/issues/1186
-module.exports = (typeof self !== 'undefined' && self.URL) || require('whatwg-url').URL;
+const URL = module.exports = (typeof self !== 'undefined' && self.URL) || require('whatwg-url').URL;
+
+/**
+ * Safely checks if the host of a URL matches a known host with a specified
+ * fallback in case of error.
+ *
+ * @param {string} url
+ * @param {string} host
+ * @param {boolean|function=} fallback
+ * @return {boolean}
+ */
+module.exports.hostMatches = function(url, host, fallback) {
+  try {
+    return new URL(url).host === host;
+  } catch (err) {
+    let result = fallback;
+    if (typeof fallback === 'function') {
```

Comment:
I feel similarly odd about this method signature. 

Here's one proposal: Keep the boolean arg, but enforce it. Then, split the callback into its own optional param.

new method signature:

```ts
  const hostMatches = function (url1: string, url2: string, recoverOnException: boolean, callback?: Function) {
    // enforce the recoverOnException boolean arg
    if (typeof recoverOnException === 'undefined')
      throw new Error('recoverOnException must be provided to hostMatches()');
    
    // ...
  }
```

with usage:
```js
    return URL.hostMatches(err.url, pageHost, true, _ => {
        debugString = 'Lighthouse was unable to determine if some API uses ' +
          `were made by this page. It's possible a Chrome extension` +
          `content script or other eval'd code is calling this API.`;
      });
```